pkg/middlewares/aws_eip_association_expander.go

```diff
@@ -0,0 +1,70 @@
+package middlewares
+
+import (
+	"github.com/cloudskiff/driftctl/pkg/resource"
+	"github.com/cloudskiff/driftctl/pkg/resource/aws"
+)
+
+/**
+  Fetching eip association from remote return every association but some of them are embedded in eip.
+  This middleware will check for every eip_association that here is no corresponding association_id inside eip.
+*/
+
+type EipAssociationExpander struct {
+	resourceFactory resource.ResourceFactory
+}
+
+func NewEipAssociationExpander(resourceFactory resource.ResourceFactory) EipAssociationExpander {
+	return EipAssociationExpander{resourceFactory}
+}
+
+func (m EipAssociationExpander) Execute(_, resourcesFromState *[]resource.Resource) error {
+	var newResources []resource.Resource
+	for _, res := range *resourcesFromState {
+		newResources = append(newResources, res)
+
+		if res.TerraformType() != aws.AwsEipResourceType {
+			continue
+		}
+		if m.haveMatchingEipAssociation(res, resourcesFromState) {
+			continue
+		}
+		// This EIP have no association, check if we need to create one
+		assocID := res.Attributes().GetString("association_id")
+		if assocID == nil || len(*assocID) <= 0 {
```

Comment:
Did you mean `*assocId == ""` ?

Reply:
yeah I mean it's empty

Reply:
I find it less intuitive than `*assocId == ""`